test/integration/action_client/active_job_assertions_test.rb

```diff
@@ -0,0 +1,226 @@
+require "test_helper"
+require "active_job_test_case"
+
+module ActionClient
+  class ActiveJobAssertionsTestCase < ActiveJobTestCase
+    include ActionClient::TestHelpers
+    MetricsClient = Class.new(ActionClient::Base) {
+      def ping(name, **query)
+        get url: "https://example.com/ping?name=#{name}", query: query
+      end
+    }
+    MetricsClientJob = Class.new(ActionClient::SubmissionJob)
+
+    def self.test(*arguments, rails: 5.2, &block)
```

Comment:
Why are some tests only run against Rails 6?

Reply:
ActiveJob assertions like `assert_enqueued_with` [_only_ accept blocks in Rails 5.2](https://api.rubyonrails.org/v5.2/classes/ActiveJob/TestHelper.html#method-i-assert_enqueued_with), and [can be invoked after the fact in Rails 6.0](https://api.rubyonrails.org/v6.0/classes/ActiveJob/TestHelper.html#method-i-assert_enqueued_with).

This test file's coverage is a mixture of both styles.

The fact that`assert_enqueued_request` relies on `assert_enqueued_with` is an implementation detail leaking out.

We could probably code around that limitation our implementation if we wanted to.

Do you think it would be a better idea to change our implementation to bridge the 5.2 to 6.0 gap, or should we do a better job of documenting how the Rails version affects the style that you can call the assertions?

Reply:
Got it. No, I don't think we should add additional complexity to support this in Rails 5.2. You could add a Note to the documentation for the Rails 6 specific features. Something as simple as "(Rails 6 or higher)" would probably be enough.